sdk/storage/README.md

```diff
@@ -1,139 +1,35 @@
-# Azure Storage Client Library for C++
+# Azure Storage libraries for C++
```

Comment:
FYI, this change broke the anchor link in the root README file that references it:
https://github.com/Azure/azure-sdk-for-cpp#getting-started

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-cpp/pull/4788